src/unix/syscall.c

```diff
@@ -1433,6 +1497,9 @@ sysreturn getrlimit(int resource, struct rlimit *rlim)
 {
     thread_log(current, "getrlimit: resource %d, rlim %p", resource, rlim);
 
+    if (rlim && !validate_user_memory(rlim, sizeof(struct rlimit), true))
+        return -EFAULT;
```

Comment:
A NULL rlim pointer is invalid and shouldn't be allowed, so I would call validate_user_memory() unconditionally, and remove `if (!rlim) return set_syscall_error(current, EINVAL);` from each of the `case` clauses below.